jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -367,6 +378,48 @@ def _parse_spectrum1d(app, file_obj, data_label=None, spectrum_viewer_reference_
     app.add_data_to_viewer(spectrum_viewer_reference_name, data_label)
 
 
+def _parse_ndarray(app, file_obj, data_label=None, data_type=None,
+                   flux_viewer_reference_name=None, spectrum_viewer_reference_name=None,
+                   uncert_viewer_reference_name=None):
+    if data_label is None:
+        data_label = app.return_data_label(file_obj)
+
+    if data_type is None:
+        data_type = 'flux'
+
+    flux = np.fliplr(np.rot90(np.moveaxis(file_obj, 0, 2), k=-1, axes=(0, 1)))
```

Comment:
`specutils` just does `np.swapaxes(0,-1)` to put the spectral axis last when it's first. Does this need to be this complicated?

And actually...do we want to do any flipping at all for the user for a bare numpy ndarray? When loading a file things get reordered because the WCS defines which axis is which, but for an ndarray it seems like maybe we should just load what the user gives us, and if they need to make it match the shape of an already loaded cube, then that's on them to do in the notebook.

Reply:
I am not sure how specutils reorder things but I needed this (similar code at two places) to accomplish the following. If you know of a simpler code that does the same thing, I'll be happy to try:

1. Baby shark must be upright and the text be readable from left to right.
2. Plain array must have `(0, 0)` at bottom left and be row-major, like how one would visualize it in `plt.imshow(array, origin='lower')`.

Reply:
And for plain array, the usual Python convention is `(z, y, x)` due to its row-major nature, and `z` is usually the spectral axis. But if you want to assume a different axis order in the input, we can also do that but we need to make it very clear in the documentation.

Reply:
Ok,  think the .gif handling is ok. The main thing I wanted to double check is that we no longer do any additional axis swapping in the `Spectrum1D` translator in `glue-astronomy`, and we don't, so that's good. My remaining concern is that the user would pull out the data from Cubeviz, do something to it, and then try to load their result in as a numpy array that already matches the shape of the loaded cube, and then this would reorder the axes to be incorrect.

@camipacifici Do you think it's reasonable from the user-side to not do any axis swapping on a plain numpy array, and leave it to the user to ensure that the shape matches the shape of previously loaded data?

Reply:
I think I found a similar problem when getting out a map from cubeviz and loading it as ndarray in imviz. It appeared swapped and I had to use np.swapaxes. I believed this came up in a help desk call too, so it is not obvious.

Reply:
Please tell me what axis should go where then. This specutils stuff really confuses me.

Reply:
I'm saying we shouldn't move any axes at all for this case in Cubeviz, we just load it as-is using `file_obj` as `flux`.

Reply:
If I simply do `flux = file_obj`, you will get this for an array of shape `(5, 8, 10)`. The order appears to be `(x, y, z)`. And the pixels are arranged in a column-major order. Is that really what you want?

![Screenshot 2022-11-03 170437](https://user-images.githubusercontent.com/2090236/199834110-c87901ea-8774-4f43-95e9-3e62d4295fec.jpg)

Reply:
If I do `flux = np.moveaxis(file_obj, 1, 0)` (something I copied from `_parse_spectrum1d_3d`), `(5, 8, 10)` becomes `(y, x, z)`. This makes it row-major (pixel increases sequentially across `x`) but assumes last axis is spectral axis, not first.

![Screenshot 2022-11-03 170804](https://user-images.githubusercontent.com/2090236/199834640-5dbe3260-6bca-482d-abca-f21775e364ee.jpg)

Which one is more correct in terms of what you expect from Cubeviz?

Reply:
I'm worried less about "user makes a brand new cube" and more about this use case (run this in the CubevizExample notebook):

```
data = cubeviz.app.get_data_from_viewer('flux-viewer', 'jw02732-o004_t004_miri_ch1-shortmediumlong_s3d.fits[SCI]')
orig_cube = data.get_object(statistic=None)
cubeviz.load_data(orig_cube.flux.value*0.1, "Test", override_cube_limit=True)
```

With the axis reordering as it is right now, this fails (well, it loads, but throws errors if you select a slice in the spectrum viewer because the axis order doesn't match the original). If you remove the reordering and just set `flux=file_obj` in the ndarray parser, this loads and matches the originally loaded cube. I feel like as a user I would expect that if I got a cube out of the app and then loaded it back in after normalizing or continuum subtracting or something, that should just work.